### PR TITLE
Javascript recognizers spanish Numbers

### PR DESCRIPTION
**Pull request by JuanAr**
Ported all numeric parser and extractor from C# to TypeScript/JavaScript implementation.

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Text.Number.Tests/TestNumberSpanish.cs [hunk 1]
```diff
@@ -156,313 +156,313 @@ public void TestNumberModel()
 
             #region Integer numbers
 
-            BasicTest(model, "2 mil millones", "2000000000");
+            //BasicTest(model, "2 mil millones", "2000000000");
```
Comment (yiylin) [inline .NET/Microsoft.Recognizers.Text.Number.Tests/TestNumberSpanish.cs]: why comment these code?

File: .NET/Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDateTimePeriodExtractorConfiguration.cs [hunk 2]
```diff
@@ -6,7 +6,7 @@ namespace Microsoft.Recognizers.Text.DateTime.Spanish
 {
     public class SpanishDateTimePeriodExtractorConfiguration : IDateTimePeriodExtractorConfiguration
     {
-        public static readonly Regex NumberCombinedWithUnit = new Regex(DateTimeDefinitions.NumberCombinedWithUnit, RegexOptions.IgnoreCase | RegexOptions.Singleline);
+        public static readonly Regex NumberCombinedWithUnit = new Regex(DateTimeDefinitions.DateTimePeriod_NumberCombinedWithUnit, RegexOptions.IgnoreCase | RegexOptions.Singleline);
```
Comment (yiylin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDateTimePeriodExtractorConfiguration.cs]: Should we follow CamelCase naming?

File: .NET/Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDurationExtractorConfiguration.cs [hunk 3]
```diff
@@ -10,7 +10,7 @@ public class SpanishDurationExtractorConfiguration : IDurationExtractorConfigura
         //TODO: improve Spanish the SuffixAndRegex
         public static readonly Regex SuffixAndRegex = new Regex(DateTimeDefinitions.SuffixAndRegex, RegexOptions.IgnoreCase | RegexOptions.Singleline);
         public static readonly Regex FollowedUnit = new Regex(DateTimeDefinitions.FollowedUnit, RegexOptions.IgnoreCase | RegexOptions.Singleline);
-        public static readonly Regex NumberCombinedWithUnit = new Regex(DateTimeDefinitions.NumberCombinedWithUnit, RegexOptions.IgnoreCase | RegexOptions.Singleline);
+        public static readonly Regex NumberCombinedWithUnit = new Regex(DateTimeDefinitions.Duration_NumberCombinedWithUnit, RegexOptions.IgnoreCase | RegexOptions.Singleline);
```
Comment (yiylin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDurationExtractorConfiguration.cs]: Should we follow CamelCase naming?

File: .NET/Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDateTimePeriodExtractorConfiguration.cs (hunk 2 above)
Comment (tellarin) [reply .NET/Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDateTimePeriodExtractorConfiguration.cs]: Not an issue. Let's keep the current form.

File: .NET/Microsoft.Recognizers.Text.Number.Tests/TestNumberSpanish.cs (hunk 1 above)
Comment (JuanAr) [reply .NET/Microsoft.Recognizers.Text.Number.Tests/TestNumberSpanish.cs]: This file was mistakenly added. We'll remove the changes from the PR.

**Comment by JuanAr** [comment]
We'll fix PR conflicts, remove the .CS changes and keep the constants camel case.

**Comment by tellarin** [comment]
I don't consider this naming an issue. These case actually make them more readable.
Let's keep the current form where some have '_' in their names.

**Pull request merged**